Src/BootCamp.Chapter/Checks.cs

```diff
@@ -1,39 +1,216 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Text;
 
 namespace BootCamp.Chapter
 {
     // This class is used to have a freedom of design, but with tests applied.
+
     public static class Checks
     {
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (Testers.IsThisStringArrayValid(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+            StringBuilder poorestName = new StringBuilder();
+            decimal amount = 0;
+            int personCount = 1;
+
+            //TODO The body of the for loop could be refactored into a function (or a few) so that it is easier to skim through the bigger function.
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var account = new Account(MakeStuffWithInput.GetNameFromString(peopleAndBalances[i]), MakeStuffWithInput.GetBalanceFromString(peopleAndBalances[i]));
+
+                CheckIfThisPersonIsPoorer(account, i);
+            }
+
+            return MakeStuffWithInput.MakefullstringPoorestPeople(personCount, poorestName.ToString(), amount);
+
+
+
+            
+            void CheckIfThisPersonIsPoorer(Account account, int i)
+            {
+                if (i == 0) // first account to be added is alway's the poorest.
+                {
+                    poorestName.Append(account.GetName());
+                    amount = account.CurrentBalance();
+                }
+                else if (amount == account.CurrentBalance()) // this account has the same balance as current.
+                {
+                    if (personCount == 1) // the second account to be added
+                    {
+                        poorestName.Append(" and " + account.GetName());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Define a constant instead of using this literal ' and ' 12 times.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5123803)